python/ray/tune/tests/test_logger.py

```diff
@@ -263,6 +266,77 @@ def testBadTBX(self):
             logger.on_trial_complete(3, [], t)
         assert "INFO" in cm.output[0]
 
+    def testAim(self):
+        config = {
+            "a": 2,
+            "b": [1, 2],
+            "c": {"c": {"D": 123}},
+            "int32": np.int32(1),
+            "int64": np.int64(2),
+            "bool8": np.bool8(True),
+            "float32": np.float32(3),
+            "float64": np.float64(4),
+            "bad": np.float128(4),
+        }
+        t = Trial(evaluated_params=config, trial_id="aim", logdir=self.test_dir)
+        logger = AimCallback()
+        logger.on_trial_result(0, [], t, result(0, 4))
+        logger.on_trial_result(1, [], t, result(1, 5))
+        logger.on_trial_result(
+            2, [], t, result(2, 6, score=[1, 2, 3], hello={"world": 1})
+        )
+
+        logger.on_trial_complete(3, [], t)
+
+        self._validate_aimresults(
+            params=(b"float32", b"float64", b"int32", b"int64", b"bool8"),
+            excluded_params=(b"bad",),
+        )
+
+    def _validate_aimresults(self, params=None, excluded_params=None):
+        try:
+            import aim
+        except ImportError:
+            print("Skipping rest of test as aim is not installed.")
+            return
+
+        print(f"{aim} aim tests here")
+        # ToDo: proper test here
+        """
+        events_file = list(glob.glob(f"{self.test_dir}/events*"))[0]
```

Comment:
Looks like we can use the [querying API](https://aimstack.readthedocs.io/en/latest/using/query_runs.html) to test it:

```python
from aim.sdk import Repo
repo = Repo(self.test_dir)

query = "metric.name == 'loss'" # Example query

# Get collection of metrics
for run_metrics_collection in my_repo.query_metrics(query).iter_runs():
    for metric in run_metrics_collection:
        # Get run params
        params = metric.run[...]
        # Get metric values
        steps, metric_values = metric.values.sparse_numpy()

        # Do some assertions here
```